Kernels/tides_kernels.py

```diff
@@ -0,0 +1,721 @@
+           "S2": {"name": "s2", "omega": 30.0000000*deg2rad*hours2s, "V_t0" : 2*T0  },
+           "N2": {"name": "n2", "omega": 28.4397295*deg2rad*hours2s, "V_t0" : 2*T0 - 3*s0 + 2*h0 + p0 },
+           "K1": {"name": "k1", "omega": 15.0410686*deg2rad*hours2s, "V_t0" : T0 + h0 - 90*deg2rad },
+           "O1": {"name": "o1", "omega": 13.9430356*deg2rad*hours2s, "V_t0" : T0 - 2*s0 + h0 + 90*deg2rad },
+           "Q1": {"name": "q1", "omega": 13.3986609*deg2rad*hours2s, "V_t0" : T0 - 3*s0 + h0 + p0 + 90*deg2rad },
+           "M4": {"name": "m4", "omega": 57.9682084*deg2rad*hours2s, "V_t0" : 4 * T0 - 4*s0 + 4*h0 },
+           "K2": {"name": "k2", "omega": 30.0821373*deg2rad*hours2s, "V_t0" : 2*T0 + 2*h0 },
+           "P1": {"name": "p1", "omega": 14.9589314*deg2rad*hours2s, "V_t0" : T0 - h0 + 90*deg2rad },
+           "Mf": {"name": "mf", "omega":  1.0980331*deg2rad*hours2s, "V_t0" : 2*s0 },
+           "Mm": {"name": "mm", "omega":  0.5443747*deg2rad*hours2s, "V_t0" : s0 - p0 }
+           }
+    
+    dimensions_Ua = {"data": "Ua", "lat": "lat", "lon": "lon"}
+    dimensions_Ug = {"data": "Ug", "lat": "lat", "lon": "lon"}
+    dimensions_Va = {"data": "Va", "lat": "lat", "lon": "lon"}
+    dimensions_Vg = {"data": "Vg", "lat": "lat", "lon": "lon"} 
+
+
+    #### select indices if you want to only load part of the dataset
+    selection = True#True#True # set to true of you only want to look at local region, use False if you want to load in part of dataset
+    if selection:
+        lat_min = 50
+        lat_max = 55
+        lon_min = -30
+        lon_max = -25
+        #FES2014 tidal fields have longitudes defined from 0...360 degrees (so -180...0 --> 180...360)
+        if lon_max < 0 :
+            lon_max+=360
+        if lon_min < 0 : 
+            lon_min+=360
+        
+        #load one file to find indices corresponding to prediscribed min and max lat and lon values
+        filenameU=os.path.join(directory_FES2014,"eastward_velocity",tides["M2"]["name"]+".nc")#
+        test_data = xr.open_dataset(filenameU) 
+        latvals = test_data["lat"].values
+        lonvals = test_data["lon"].values
+        #put lon and lat valus on grid       
+        latvals_grid, lonvals_grid = np.meshgrid(latvals,lonvals) 
+        
+        #find indices closest to prediscribed lat and longitudes
+        iy_min, ix_min = getclosest_ij(latvals_grid, lonvals_grid, lat_min,lon_min)
+        iy_max, ix_max = getclosest_ij(latvals_grid, lonvals_grid, lat_max, lon_max)
+        ix_min -= 1
+        ix_max += 1
+        iy_min -= 1
+        iy_max += 1
+        if(lon_min<lon_max):  #selection of longitudes does not cross end of FES2014 grid
+            indices = {"lat": range(ix_min, ix_max), "lon": range(iy_min, iy_max)}
+        else: #selection of longitudes does cross end of FES2014 grid
+            iy_end = lonvals.size
+            indices = {"lat": range(ix_min, ix_max),"lon": list(range(0, iy_max)) + list(range(iy_min,iy_end))}
+    else:
```

Comment:
Not sure this bit is needed? Not essential for the working of the script

Reply:
No it is not needed indeed if you load the entire dataset. I only added it to help select the same indices when you want to only load an area, given that the FES2014 datet has the longitudes from 0 to 360 instead of -180 to 180.